R/geom-label-repel.R

```diff
@@ -211,14 +211,30 @@ GeomLabelRepel <- ggproto(
 #' @noRd
 makeContent.labelrepeltree <- function(x) {
 
+  # Absolute panel size
+  width  <- width_cm( unit(1, "npc"))
+  height <- height_cm(unit(1, "npc"))
+
+  # Translate points to cm (assumed to be 0-1 range native at first)
+  x$data$x <- x$data$x * width
+  x$data$y <- x$data$y * height
+
   # The padding around each bounding box.
-  box_padding_x <- convertWidth(x$box.padding, "npc", valueOnly = TRUE)
-  box_padding_y <- convertHeight(x$box.padding, "npc", valueOnly = TRUE)
+  box.padding   <- length_cm(x$box.padding)
+  label.padding <- length_cm(x$label.padding)
+
+  # Input point diameter is assumed to be in mm
+  # We convert to radius in centimetres, by dividing by 20
+  # `.pt / .stroke` ~= 0.75 accounts for a historical error in ggplot2
+  point.size <- x$data$point.size
+  point.size[is.na(point.size)] <- 0
+  point.size <- point.size * .pt / .stroke / 20
```

Comment:
I  think this here was the magic incantation you were looking for in https://github.com/slowkow/ggrepel/issues/83#issuecomment-547221120

Reply:
Amazing, I don't think I would have ever figured this out by myself. 🙏

You're a wizard 🧙‍♂️